### PR TITLE
Simplify adding network

### PR DESCRIPTION
**Pull request by Tbaut**
I need more networks to test.
Westend and Fast Westend will be visible only in dev in the dropdown

I kept an `if` when we're asking for light client, because the dynamic import needs to have a deterministic name. Otherwise we'll load the descriptor somewhat dynamically.

If we want to add a network now, we can add it to the `networks.json` and follow the TS errors, more or less everything is in the network context beside the menu (in `header.tsx`).

### CODE REVIEW

File: src/contexts/NetworkContext.tsx [hunk 1]
```diff
@@ -36,58 +54,43 @@ const NetworkContextProvider = ({ children }: NetworkContextProps) => {
...
+        relayChain = import('polkadot-api/chains/ksmcc3').then(
           ({ chainSpec }) => smoldot.addChain({ chainSpec }),
         )
-        cl = createClient(getSmProvider(ksmRelayChain))
-        typedApi = cl.getTypedApi(ksm)
-        break
       }
+      //@ts-expect-error the Chain type isn't exported
+      //it comes from 'smoldot' that we don't import
+      client = createClient(getSmProvider(relayChain as unknown))
```
Comment (wirednkod) [inline src/contexts/NetworkContext.tsx:81]: Smoldot can be imported from polkadot-api as well so you can add at the top:
```ts
import type { Chain } from 'polkadot-api/smoldot'
```
and then:
```suggestion
      client = createClient(getSmProvider(relayChain as Chain))
```

File: src/header.tsx [hunk 2]
```diff
@@ -23,20 +23,29 @@ import { PanelLeft } from 'lucide-react'
...
-  'Polkadot|polkadot',
-  'Polkadot Light Client|polkadot-lc',
-  'Kusama|kusama',
-  'Kusama Light Client|kusama-lc',
+const networkList: { name: SupportedNetworkNames; display: string }[] = [
+  { name: 'polkadot', display: 'Polkadot' },
+  { name: 'polkadot-lc', display: 'Polkadot Light Client' },
+  { name: 'kusama', display: 'Kusama' },
+  { name: 'kusama-lc', display: 'Kusama Light Client' },
 ]
```
Comment (wirednkod) [inline src/header.tsx:33]:
```suggestion
type NetworkListType =  { name: SupportedNetworkNames; display: string }

const networkList: NetworkListType[] = [
  { name: 'polkadot', display: 'Polkadot' },
  { name: 'polkadot-lc', display: 'Polkadot Light Client' },
  { name: 'kusama', display: 'Kusama' },
  { name: 'kusama-lc', display: 'Kusama Light Client' },
]

(import.meta.env.DEV)  &&  networkList.push(
	{ name: 'westend', display: 'Westend' },
        { name: 'fast-westend', display: 'Fast Westend' }
        )
```

File: src/header.tsx [hunk 3]
```diff
@@ -23,20 +23,29 @@ import { PanelLeft } from 'lucide-react'
...
 
+  useEffect(() => {
+    if (import.meta.env.DEV) {
+      networkList.push(
+        { name: 'westend', display: 'Westend' },
+        { name: 'fast-westend', display: 'Fast Westend' },
+      )
+    }
+  }, [])
+
```
Comment (wirednkod) [inline src/header.tsx:48]: Better to add this outside the  component to reduce render time and actions (see above)

```suggestion
```

File: src/header.tsx [hunk 4]
```diff
@@ -94,13 +103,13 @@ export const Header = () => {
               </Button>
             </DropdownMenuTrigger>
             <DropdownMenuContent>
-              {networkList.map((n) => (
+              {networkList.map(({ name, display }) => (
```
Comment (wirednkod) [inline src/header.tsx:106]:
```suggestion
              {networkList.map(({ name, display }: NetworkListType) => (
```

**Review by wirednkod — CHANGES_REQUESTED** [review]

File: src/contexts/NetworkContext.tsx (hunk 1 above)
Comment (Tbaut) [reply src/contexts/NetworkContext.tsx:81]: oh nice, somehow my IDE didn't see it

File: src/header.tsx (hunk 4 above)
Comment (Tbaut) [reply src/header.tsx:107]: it's inferred now, gave a name to this type.

**Pull request merged**